ocaml/typing/typecore.ml

```diff
@@ -5292,43 +5181,28 @@ and type_expect_
   | Pexp_function _ ->
       Misc.fatal_error "non-Jane-Syntax [Pexp_function] made it to typechecking"
   | Pexp_apply
-      ({ pexp_desc = Pexp_extension({
-            txt = ("ocaml.unique" | "unique" | "extension.unique" as txt)}, PStr []) },
-       [Nolabel, sbody]) ->
-      if txt = "extension.unique" && not (Language_extension.is_enabled Unique) then
-        raise (Typetexp.Error (loc, Env.empty, Unsupported_extension Unique));
-      let expected_mode = mode_unique expected_mode in
-      let expected_mode = expect_mode_cross env ty_expected expected_mode in
-      let exp =
-        type_expect ~recarg env expected_mode sbody
-          ty_expected_explained
-      in
-      {exp with exp_loc = loc}
-  | Pexp_apply
-      ({ pexp_desc = Pexp_extension({
-          txt = ("ocaml.once" | "once" | "extension.once" as txt)}, PStr []) },
-      [Nolabel, sbody]) ->
-      if txt = "extension.once" && not (Language_extension.is_enabled Unique) then
-        raise (Typetexp.Error (loc, Env.empty, Unsupported_extension Unique));
-      let expected_mode = expect_mode_cross env ty_expected expected_mode in
-      submode ~loc ~env ~reason:Other
-        (Value.min_with_linearity Linearity.once) expected_mode;
-      let expected_mode = mode_once expected_mode in
-      let exp =
-        type_expect ~recarg env expected_mode sbody
-          ty_expected_explained
+      ({ pexp_desc = Pexp_extension({txt; _}, payload); pexp_loc},
+        [Nolabel, sbody]) when txt = Jane_syntax.Mode_expr.embedded_name_str ->
+      let modes = Jane_syntax.Mode_expr.of_payload ~loc:pexp_loc payload in
+      let expected_mode = List.fold_left (fun expected_mode mode ->
+          match mode.txt with
+          | "unique" ->
+            let expected_mode = mode_unique expected_mode in
+            expect_mode_cross env ty_expected expected_mode
+          | "once" ->
+            let expected_mode = expect_mode_cross env ty_expected expected_mode in
+            submode ~loc ~env ~reason:Other
+              (Value.min_with_linearity Linearity.once) expected_mode;
+            mode_once expected_mode
+          | "local" ->
+            let expected_mode = expect_mode_cross env ty_expected expected_mode in
+            submode ~loc ~env ~reason:Other
+              (Value.min_with_locality Regionality.local) expected_mode;
+            mode_strictly_local expected_mode
+          | _ ->
+            raise (Error (mode.loc, env, Invalid_mode_coercion))
```

Comment:
Isn't this case unreachable? It's a bit inconsistent to raise a proper error here but to raise a fatal error (assert false) in typemexp.ml. Can we be consistent?